pkg/workers/worker_interface_test.go

```diff
@@ -0,0 +1,362 @@
+package workers
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/signalbus"
+	"github.com/onsi/gomega"
+)
+
+func TestBaseWorker_GetID(t *testing.T) {
+	type fields struct {
+		Id string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "should return the id of the base worker",
+			fields: fields{
+				Id: "base_worker_id",
+			},
+			want:    "base_worker_id",
+			wantErr: false,
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			t.Parallel()
+			b := &BaseWorker{
+				Id: tt.fields.Id,
+			}
+			g.Expect(b.GetID()).To(gomega.Equal(tt.want))
```

Comment:
Most of the tests here show that a getter returns the value set a couple of lines above.
I'm not sure it makes sense to test getter/setters.
IMO, the only useful tests here are the tests for `StartWorker` and `StopWorker`

@JasonPowr @machi1990 @pawelpaszki WDYT?

Reply:
I am on two minds here. getters/ setters should probably be tested if they are complex enough (rather then just setting/ getting values from some constants). I don't have a strong preference TBH. since the tests area already written I guess there is no harm in keeping them. WDYT?